utils/global-state-update-gen/src/utils/account_manager.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::BTreeMap;
+
+use casper_engine_test_support::LmdbWasmTestBuilder;
+use casper_types::{
+    account::Account, AccessRights, CLValue, Key, PublicKey, StoredValue, URef, U512,
+};
+
+use rand::prelude::*;
+
+use crate::utils::print_entry;
+
+pub struct AccountManager {
+    accounts: BTreeMap<PublicKey, Account>,
+}
+
+impl AccountManager {
+    /// Creates a new instance of `AccountManager`.
+    pub fn new() -> Self {
+        Self {
+            accounts: Default::default(),
+        }
+    }
+
+    /// Gets the account for the given public key.
+    /// If the account doesn't exist, it prints the entries necessary to create it and returns the
+    /// created account.
+    pub fn get_or_create_account(
+        &mut self,
+        builder: &mut LmdbWasmTestBuilder,
+        pkey: &PublicKey,
+    ) -> &Account {
+        // unfortunately, `if let Some(account) = self.accounts.get(&pkey)` upsets the borrow
+        // checker, so we have to do it this way
+        if !self.accounts.contains_key(pkey) {
+            let account_hash = pkey.to_account_hash();
+            if let Some(account) = builder.get_account(account_hash) {
+                let _ = self.accounts.insert(pkey.clone(), account);
+                // unwrapping is safe, we just inserted it
+                self.accounts.get(pkey).unwrap()
+            } else {
+                self.create_account(pkey.clone());
+                // create_account inserts the account, so unwrapping is safe
+                self.accounts.get(pkey).unwrap()
+            }
+        } else {
+            // we know it contains the key, unwrapping safe
+            self.accounts.get(pkey).unwrap()
+        }
+    }
+
+    fn create_account(&mut self, pkey: PublicKey) {
+        let mut rng = rand::thread_rng();
+        let new_purse = URef::new(rng.gen(), AccessRights::READ_ADD_WRITE);
+
+        // Apparently EE does this, not sure why, but we'll also do it for consistency.
```

Comment:
All purses' URef's point to `()` so they can't be modified directly by the purse owner.